packages/os/bootstrap-containers-toml

```diff
@@ -0,0 +1,16 @@
+[required-extensions]
+bootstrap-containers = "v1"
++++
+{{#if settings.bootstrap-containers}}
+{{#each settings.bootstrap-containers}}
```

Comment:
Wouldn't it be more readable (and probably more maintainable in the long run) to have this rendered by a handlebars helper? e.g.

```hbs
{{bootstrap-containers settings.bootstrap-containers}}
```

Then you can just use Rust and strong typing to generate the toml tables. . I want us to avoid another "generate on-the-flight complex structures" and use the Rust to guarantee the correctness of the serialized format. (@bcressey what do you think?)

Reply:
I like this approach, this could just be a generic `settings-to-toml` helper which we could use for all of these PRs where we substitute reading from the API for reading from a config file

Reply:
> Wouldn't it be more readable (and probably more maintainable in the long run) to have this rendered by a handlebars helper?

It doesn't really help in the long run where the goal is to decouple templates and settings extensions so they can be maintained independently.

> I like this approach, this could just be a generic settings-to-toml helper which we could use for all of these PRs where we substitute reading from the API for reading from a config file

I wasn't too keen on `settings-to-json` when [that put in an appearance](https://github.com/bottlerocket-os/bottlerocket/pull/3834#discussion_r1534882924) recently.

I am a little more positively inclined on something like `settings-to-toml` if the helper is producing a 1:1 match with a subset of the external system settings API, if only because that API is covered by a strong covenant with the end user that it won't change.

However I'm still not very enthusiastic. I'd like most of these agents to be able to be configured in ways that aren't 100% reflected in the external API - like `kubelet` and `ecs-agent` are not - since some behaviors may make sense to hard-code in certain variants.

Ultimately an agent whose config is nothing but the external API is in a precarious position: it makes more sense to me to have use a combination of Bottlerocket features and systemd functionality to dispense with that agent entirely (as with `ecs-settings-applier` in the linked PR) where we can, than to add helpers which special-case just that agent rather than answering a more general need.